src/main/java/org/spongepowered/api/event/entity/AffectEntityEvent.java

```diff
@@ -109,17 +98,5 @@
      * @param predicate The predicate to use for filtering
      * @return The entities removed from {@link #entities()}
      */
-    default List<? extends Entity> filterEntities(Predicate<Entity> predicate) {
-        final List<Entity> removedEntities = new ArrayList<>();
-
-        final Iterator<Entity> i = this.entities().iterator();
-        while (i.hasNext()) {
-            final Entity entity = i.next();
-            if (!predicate.test(entity)) {
-                i.remove();
-                removedEntities.add(entity);
-            }
-        }
-        return removedEntities;
-    }
+    List<Entity> filterEntities(Predicate<Entity> predicate);
```

Comment:
To optimize I suggest returning void. I doubt many plugins need this list. For the few ones that need it, elements can still be collected in the predicate.

Reply:
I'm with you there I think. I dunno if @gabizou or @Zidane have an opposing view, but in general I think plugins are going to care about what _is_ going to be affected, not what _isn't_. I think we provide the original list of entities that are affected anyway, right? Plugins could also diff that if they need all entities affected by the event regardless.

Reply:
There is `List<EntitySnapshot> entitySnapshots()` which is immutable but must be called at `Order#PRE` to be initialized.